tools/importer-msgraph-metadata/pipeline/task_template_operations.go

```diff
@@ -0,0 +1,300 @@
+	if operation.ResourceId != nil {
+		resourceIdCode = fmt.Sprintf(`new %sId()`, operation.ResourceId.Name)
+	}
+
+	uriSuffixCode := "null"
+	if operation.UriSuffix != nil {
+		uriSuffixCode = fmt.Sprintf(`"%s"`, *operation.UriSuffix)
+	}
+
+	responseObjectCode := "null"
+	if responseModel != nil {
+		responseObjectCode = fmt.Sprintf("typeof(%sModel)", *responseModel)
+	} else if responseType != nil {
+		responseObjectCode = fmt.Sprintf("typeof(%s)", *responseType)
+	}
+
+	return fmt.Sprintf(`using Pandora.Definitions.CustomTypes;
+using Pandora.Definitions.Interfaces;
+%[2]s
+using System.Collections.Generic;
+using System.Net;
+using System;
+
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+namespace %[1]s;
+
+internal class %[3]sOperation : Operations.%[4]sOperation
+{
+%[9]s
+    public override IEnumerable<HttpStatusCode> ExpectedStatusCodes() => new List<HttpStatusCode>
+        {
+%[5]s,
+        };
+    public override ResourceID? ResourceId() => %[6]s;
+    public override Type? ResponseObject() => %[10]s;
+    public override string? UriSuffix() => %[8]s;
+}
+`, namespace, modelsImportCode, operation.Name, strings.Title(strings.ToLower(operation.Method)), expectedStatusesCode, resourceIdCode, responseModel, uriSuffixCode, contentTypeCode, responseObjectCode)
+}
+
+func templateCreateUpdateOperation(namespace, modelsNamespace string, operation *Operation, contentType, responseModel *string, statuses []string) string {
+	modelsImportCode := ""
+	if modelsNamespace != "" {
+		modelsImportCode = fmt.Sprintf("using %s;", modelsNamespace)
+	}
+
+	statusEnums := make([]string, len(statuses))
+	for i, status := range statuses {
+		code, _ := strconv.Atoi(status)
+		statusEnums[i] = csHttpStatusCode(code)
+	}
+
+	expectedStatusesCode := indentSpace(strings.Join(statusEnums, ",\n"), 16)
+
+	contentTypeCode := ""
+	if contentType != nil && !strings.HasPrefix(*contentType, "application/json") {
+		contentTypeCode = indentSpace(fmt.Sprintf(`public override string? ContentType() => "%s";`, *contentType), 4)
+	}
+
+	resourceIdCode := "null"
+	if operation.ResourceId != nil {
+		resourceIdCode = fmt.Sprintf(`new %sId()`, operation.ResourceId.Name)
+	}
+
+	uriSuffixCode := "null"
+	if operation.UriSuffix != nil {
+		uriSuffixCode = fmt.Sprintf(`"%s"`, *operation.UriSuffix)
+	}
+
+	requestObjectCode := "null"
+	if operation.RequestModel != nil {
+		requestObjectCode = fmt.Sprintf("typeof(%sModel)", *operation.RequestModel)
+	} else if operation.RequestType != nil {
+		requestObjectCode = fmt.Sprintf("typeof(%s)", *operation.RequestType.CSType())
+	}
+
+	responseObjectCode := "null"
+	if responseModel != nil {
+		responseObjectCode = fmt.Sprintf("typeof(%sModel)", *responseModel)
+	}
+
+	return fmt.Sprintf(`using Pandora.Definitions.CustomTypes;
+using Pandora.Definitions.Interfaces;
+%[2]s
+using System.Collections.Generic;
+using System.Net;
+using System;
+
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
```

Comment:
fwiw it might be worth reordering this, since `copywrite` only checks the first N bytes of a file

```suggestion
	return fmt.Sprintf(`// Copyright (c) HashiCorp, Inc.
// SPDX-License-Identifier: MPL-2.0

using Pandora.Definitions.CustomTypes;
using Pandora.Definitions.Interfaces;
%[2]s
using System.Collections.Generic;
using System.Net;
using System;
```